targets/mbedos5/jerryscript-mbed/jerryscript-mbed-drivers/source/DigitalOut.cpp

```diff
@@ -0,0 +1,131 @@
+/* Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "jerryscript-mbed-util/logging.h"
+#include "jerryscript-mbed-library-registry/wrap_tools.h"
+
+#include "mbed.h"
+
+/**
+ * DigitalOut#write (native JavaScript method)
+ *
+ * Writes a binary value to a DigitalOut.
+ *
+ * @param value 1 or 0, specifying whether the output pin is high or low,
+ *      respectively
+ * @returns undefined, or an error if invalid arguments are provided.
+ */
+DECLARE_CLASS_FUNCTION(DigitalOut, write) {
+    CHECK_ARGUMENT_COUNT(DigitalOut, write, (args_count == 1));
+    CHECK_ARGUMENT_TYPE_ALWAYS(DigitalOut, write, 0, number);
+
+    // Extract native DigitalOut pointer
+    uintptr_t ptr_val;
+    jerry_get_object_native_handle(this_obj, &ptr_val);
+
+    DigitalOut* native_ptr = reinterpret_cast<DigitalOut*>(ptr_val);
+    
+    int arg0 = jerry_get_number_value(args[0]);
+    native_ptr->write(arg0);
+
+    return jerry_create_undefined();
+}
+
+/**
+ * DigitalOut#read (native JavaScript method)
+ *
+ * Reads the current status of a DigitalOut
+ *
+ * @returns 1 if the pin is currently high, or 0 if the pin is currently low.
+ */
+DECLARE_CLASS_FUNCTION(DigitalOut, read) {
+    CHECK_ARGUMENT_COUNT(DigitalOut, read, (args_count == 0));
+
+    // Extract native DigitalOut pointer
+    uintptr_t ptr_val;
+    jerry_get_object_native_handle(this_obj, &ptr_val);
+
+    DigitalOut* native_ptr = reinterpret_cast<DigitalOut*>(ptr_val);
+
+    int result = native_ptr->read();
+    return jerry_create_number(result);
+}
+
+/**
+ * DigitalOut#is_connected (native JavaScript method)
+ *
+ * @returns 0 if the DigitalOut is set to NC, or 1 if it is connected to an
```

Comment:
Is the 0/1 return value intentional? JS does have a boolean type. Why not return true/false?
(Not insisting though, just asking.)